### PR TITLE
Pass on definitions flags properly

### PR DESCRIPTION
**Pull request by mfherbst**
@maxscheurer You were a bit too brutal with removing stuff. Libtensor wants a lot more flags to be set during compilation unfortunately. Some even bleed into the interfaces. I updated the pc, but did not bother with the cmake. Feel free to do that or otherwise kick that mechanism.

### CODE REVIEW

**Comment by maxscheurer** [comment]
I'm surprised that this did not show up, because I was able to use lt via CMake seamlessly.

**Comment by maxscheurer** [comment]
And I did not try via pkgconfig of course 🙈

**Comment by maxscheurer** [comment]
I think it's because I was using CMake for `adccore`, where the threading libs etc. were "found" again, so it did not show up in this workflow.

**Comment by mfherbst** [comment]
Yes that could be. Anyway ... I'm close to having a working libadcc again.

**Comment by maxscheurer** [comment]
I removed the CMake config now.

**Review by maxscheurer — APPROVED** [review]

**Comment by mfherbst** [comment]
As always things are not all that easy. Now I get a:
```
libtensor::product_table_container::req_table(const id_t&), ../libtensor/symmetry/product_table_container.C (86), bad_parameter
Table does not exist.
```
when running the adcc tests.

**Comment by maxscheurer** [comment]
This looks like something we had in early 2019... it rings a bell in the context of `pybind11` and `-fvisibility=hidden`... if that helps?

**Comment by mfherbst** [comment]
yes that might help, actually. I'll try to investigate that.

**Comment by maxscheurer** [comment]
BTW the version in `setup.cfg` and `CMakeLists.txt` are not the same.

**Comment by mfherbst** [comment]
Wow ... that seems to have solved it indeed. Good memory! Now I just have to do this a bit more fine granular and understand why.

**Comment by mfherbst** [comment]
oh that should not be.

**Comment by mfherbst** [comment]
We must still be doing something very wrong with the way we setup libtensor, because the adcc compiled against this is about 4 times as slow.

File: build_libtensor.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        args += ["-DCMAKE_CXX_FLAGS_DEBUG={}".format(cpflags),
+                 "-DCMAKE_C_FLAGS_DEBUG={}".format(cpflags),
+                 "-DCMAKE_EXE_LINKER_FLAGS_DEBUG=\"{}\"".format(ldflags),
+                 "-DCMAKE_BUILD_TYPE=Debug"]
+    elif build_type:
+        raise SystemExit("Unknown build type: " + build_type)
+
+    if "mpi" in features:
+        args += ["-DWITH_MPI=ON"]
+    if "libxc" in features:
```
Comment (maxscheurer) [inline build_libtensor.py]: I know you do DFT now... but this needs to be libxm 😁

**Comment by maxscheurer** [comment]
> We must still be doing something very wrong with the way we setup libtensor, because the adcc compiled against this is about 4 times as slow.

Oh boy... I did not expect this.
That's frustrating. I hope it's not because of the older version as such 😰🙄

File: build_libtensor.py (hunk 1 above)
Comment (mfherbst) [reply build_libtensor.py]: Getting tired you know ;).

**Comment by mfherbst** [comment]
I don't think so. The threading profile is visibly very different. So there is more to it.

**Comment by mfherbst** [comment]
Ok. I give up for now. I'm starting to get quite frustrated over this I have to say ...

**Comment by maxscheurer** [comment]
I can imagine... Really frustrating. I hope I can help maybe tomorrow 👍🏻

**Pull request merged**